src/library/clementine/clementinedbconnection.cpp

```diff
@@ -0,0 +1,170 @@
+
+    if (query.exec()) {
+        while (query.next()) {
+            playlist.playlistId = query.value(0).toString();
+            playlist.name = query.value(1).toString();
+            list.append(playlist);
+        }
+    } else {
+        LOG_FAILED_QUERY(query);
+    }
+    return list;
+}
+
+QList<struct ClementineDbConnection::PlaylistEntry> ClementineDbConnection::getPlaylistEntries(int playlistId) {
+
+    PerformanceTimer time;
+    time.start();
+
+    QList<struct ClementineDbConnection::PlaylistEntry> list;
+    struct ClementineDbConnection::PlaylistEntry entry;
+
+    QSqlQuery query(m_database);
+    query.setForwardOnly(true); // Saves about 50% time
+
+    QString queryString;
+
+	queryString = QString(
+		"SELECT "
+		"ROWID, "
+		"playlist_items.title, "              // 1
+		"playlist_items.filename, "           // 2
+		"playlist_items.length, "             // 3
+		"playlist_items.artist, "             // 4
+		"playlist_items.year, "               // 5
+		"playlist_items.album, "              // 6
+		"playlist_items.rating, "             // 7
+		"playlist_items.genre, "              // 8
+		"playlist_items.track, "              // 9
+		//"playlist_items.DateAddedStamp, "   //
+		"playlist_items.bpm, "                // 10
+		"playlist_items.bitrate, "            // 11
+		"playlist_items.comment, "            // 12
+		"playlist_items.playcount, "          // 13
+		"playlist_items.composer, "           // 14
+		"playlist_items.grouping, "           // 15
+		"playlist_items.type, "           // 16
+		"playlist_items.albumartist "           // 17
+		"FROM playlist_items "
+		"WHERE playlist_items.playlist = %1")
+			.arg(playlistId);
+
+
+    query.prepare(queryString);
+
+    if (query.exec()) {
+        while (query.next()) {
+        	QString type =  query.value(16).toString();
+        	if(type != "File")
+        		continue;
+            entry.artist = query.value(4).toString();
+            entry.title = query.value(1).toString();
+            entry.trackId = query.value(0).toInt();
+            entry.uri = QUrl::fromEncoded(query.value(2).toByteArray(), QUrl::StrictMode);
+            entry.duration = int(query.value(3).toLongLong() /1000000000);
+            entry.year = query.value(5).toInt();
+            entry.album = query.value(6).toString();
+            entry.albumartist = query.value(17).toString();
+
+            entry.rating = query.value(7).toInt();
+            entry.genre = query.value(8).toString();
+            entry.grouping = query.value(15).toString();
+            entry.tracknumber = query.value(9).toInt();
+            entry.bpm = query.value(10).toInt();
+            entry.bitrate = query.value(11).toInt();
+            entry.comment = query.value(12).toString();
+            entry.playcount = query.value(13).toInt();
+            entry.composer = query.value(14).toString();
+
+            if(entry.artist == "" && entry.title == ""){
+            	entry.artist = "Unknown";
```

Comment:
Im afraid that "Unknown" will be finally witten to the file tags if we replace it here.
Why do we need it?

Reply:
There are some entrys in the clementine database which get corrupted in a way that most of the informations about the track is missing, even if the filedirectory is still valid and the entrys work in Clementine. I dont know the reason for that, but i was choosing the way to just use the filename as tracktitle and "unknown" as Artist, for better readability and completeness of the playlist, even if the track is not valid anymore in the clementine playlist.
As far as i understood, the import of the playlists are just temporary in the mixxx-internal database for each startup and no information is written back to the id tags. Am i right?